pkg/apis/apps/v1beta1/defaults.go

```diff
@@ -30,6 +30,10 @@ func SetDefaults_StatefulSet(obj *StatefulSet) {
 	if len(obj.Spec.PodManagementPolicy) == 0 {
 		obj.Spec.PodManagementPolicy = OrderedReadyPodManagement
 	}
+
+	if obj.Spec.UpdateStrategy.Type == "" {
+		obj.Spec.UpdateStrategy.Type = RollingUpdateStatefulSetStrategyType
```

Comment:
I'm fine with defaulting to OnDelete. The plan would be to maintain this for the lifetime of v1beta1 an then move it to RollingUpdate in GA (or whatever the next API version we support is)?